tests/playbooks/roles/install-csi-cinder/tasks/main.yaml

```diff
@@ -0,0 +1,206 @@
+        name: cluster-admin
+      subjects:
+      - kind: ServiceAccount
+        name: cloud-controller-manager
+        namespace: kube-system
+      ---
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: openstack-cloud-controller-manager
+        namespace: kube-system
+        labels:
+          k8s-app: openstack-cloud-controller-manager
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            k8s-app: openstack-cloud-controller-manager
+        template:
+          metadata:
+            labels:
+              k8s-app: openstack-cloud-controller-manager
+          spec:
+            tolerations:
+            - key: node.cloudprovider.kubernetes.io/uninitialized
+              value: "true"
+              effect: NoSchedule
+            - key: node-role.kubernetes.io/master
+              effect: NoSchedule
+            serviceAccountName: cloud-controller-manager
+            imagePullSecrets:
+            - name: openlab
+            containers:
+              - name: openstack-cloud-controller-manager
+                image: docker.io/k8scloudprovider/openstack-cloud-controller-manager:latest
+                args:
+                  - /bin/openstack-cloud-controller-manager
+                  - --v=4
+                  - --cloud-config=/etc/config/cloud.conf
+                  - --cloud-provider=openstack
+                  - --use-service-account-credentials=true
+                  - --bind-address=127.0.0.1
+                volumeMounts:
+                  - mountPath: /etc/config
+                    name: cloud-config-volume
+                    readOnly: true
+            hostNetwork: true
+            volumes:
+            - name: cloud-config-volume
+              secret:
+                secretName: cloud-config
+      EOF
+
+- name: Wait for openstack-cloud-controller-manager up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep openstack-cloud-controller-manager | grep Running
+  register: check_occm
+  until: check_occm.rc == 0
+  retries: 24
+  delay: 5
+  ignore_errors: yes
+
+- name: Deploy cinder-csi-plugin
+  shell:
+    executable: /bin/bash
+    cmd: |
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+      kubectl apply -f manifests/cinder-csi-plugin
+
+- name: Wait for csi-cinder-controllerplugin up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep csi-cinder-controllerplugin  | grep Running
+  register: check_csi
+  until: check_csi.rc == 0
+  retries: 24
+  delay: 5
+
+- name: Wait for csi-cinder-nodeplugin up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep csi-cinder-nodeplugin  | grep Running
+  register: check_csi
+  until: check_csi.rc == 0
+  retries: 24
+  delay: 5
+
+- name: Gather additional evidence if csi-cinder-plugin failed to come up
+  when: check_csi.failed
```

Comment:
If we want to print error message here, we should ignore the error in above 2 tasks.